src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -947,6 +947,1128 @@ public static class Extensions
+                        IL_0017: ldarg.0
+                        IL_0018: ldfld int32 IntExt/'<>c__DisplayClass1_0'::b
+                        IL_001d: call void [netstandard]System.Console::WriteLine(int32)
+                        IL_0022: ret
+                    } // end of method '<>c__DisplayClass1_0'::'<Foo>b__0'
+                } // end of class <>c__DisplayClass1_0
+                // Methods
+                .method public hidebysig specialname static 
+                    class [netstandard]System.Action Foo () cil managed 
+                {
+                    // Method begins at RVA 0x207c
+                    // Code size 52 (0x34)
+                    .maxstack 3
+                    .locals init (
+                        [0] class [netstandard]System.Action,
+                        [1] int32
+                    )
+                    IL_0000: newobj instance void IntExt/'<>c__DisplayClass1_0'::.ctor()
+                    IL_0005: dup
+                    IL_0006: ldc.i4.7
+                    IL_0007: stfld int32 IntExt/'<>c__DisplayClass1_0'::b
+                    IL_000c: dup
+                    IL_000d: ldftn instance void IntExt/'<>c__DisplayClass1_0'::'<Foo>b__0'()
+                    IL_0013: newobj instance void [netstandard]System.Action::.ctor(object, native int)
+                    IL_0018: stloc.0
+                    IL_0019: ldstr "Foo()"
+                    IL_001e: call void [netstandard]System.Console::WriteLine(string)
+                    IL_0023: dup
+                    IL_0024: ldfld int32 IntExt/'<>c__DisplayClass1_0'::b
+                    IL_0029: ldc.i4.1
+                    IL_002a: add
+                    IL_002b: stloc.1
+                    IL_002c: ldloc.1
+                    IL_002d: stfld int32 IntExt/'<>c__DisplayClass1_0'::b
+                    IL_0032: ldloc.0
+                    IL_0033: ret
+                } // end of method IntExt::Foo
+            } // end of class IntExt
+            """
+        );
+
+        var tree = comp.SyntaxTrees[0];
+        var model = comp.GetSemanticModel(tree);
+        var extension1 = tree.GetRoot().DescendantNodes().OfType<ExtensionDeclarationSyntax>().First();
+        var symbol1 = model.GetDeclaredSymbol(extension1);
+        var sourceExtension1 = symbol1.GetSymbol<SourceNamedTypeSymbol>();
+        Assert.Equal("<>E__0", symbol1.MetadataName);
+        Assert.Equal("IntExt.<>E__0", symbol1.ToTestDisplayString());
+    }
+
+    [Fact]
+    [WorkItem("https://github.com/dotnet/roslyn/issues/78042")]
+    public void ExtensionWithCapturing3()
+    {
+        var src = """
+using System.Runtime.CompilerServices;
+using System.Collections.Generic;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+Console.WriteLine("Hello, World!");
+
+namespace System.Linq
+{
+    public static partial class AsyncEnumerable
+    {
+        extension<TSource>(IAsyncEnumerable<TSource> source)
+        {
+            /// <summary>
```

Comment:
>/// <summary> [](http://example.com/codeflow?start=12&length=13)

Let's reduce the size of the string and remove all the comments that are not related to the issue we are testing.